code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -69,14 +71,41 @@
 		apply_assigned_role_to_spawn(ai_job)
 
 	GLOB.ai_list += src
-
+	var/datum/atom_hud/H = GLOB.huds[DATA_HUD_SQUAD]
+	H.add_hud_to(src)
+
+	RegisterSignal(src, COMSIG_MOB_CLICK_ALT, .proc/send_order)
+	RegisterSignal(src, COMSIG_ORDER_SELECTED, .proc/set_order)
+
+	var/datum/action/innate/order/attack_order/send_attack_order = new
+	var/datum/action/innate/order/defend_order/send_defend_order = new
+	var/datum/action/innate/order/retreat_order/send_retreat_order = new
+	send_attack_order.target = src
+	send_attack_order.give_action(src)
+	send_defend_order.target = src
+	send_defend_order.give_action(src)
+	send_retreat_order.target = src
+	send_retreat_order.give_action(src)
 
 /mob/living/silicon/ai/Destroy()
 	GLOB.ai_list -= src
 	QDEL_NULL(builtInCamera)
 	QDEL_NULL(track)
+	UnregisterSignal(src, COMSIG_ORDER_SELECTED)
+	UnregisterSignal(src, COMSIG_MOB_CLICK_ALT)
 	return ..()
 
+///Print order visual to all marines squad hud and give them an arrow to follow the waypoint
+/mob/living/silicon/ai/proc/send_order(datum/source, atom/target)
+	SIGNAL_HANDLER
+	if(!current_order)
+		to_chat(src, "<span class='warning'>Your have no order selected.</span>")
+	current_order.send_order(target)
+
+///Set the current order
+/mob/living/silicon/ai/proc/set_order(datum/source, datum/action/innate/order/order)
+	SIGNAL_HANDLER
+	current_order = order
```

Comment:
Why are you using signals as setters

Reply:
component base architecture?

Reply:
The selecting and unselecting is done order side. Since orders are generic, the owner of an order is just a mob and doesn't have a current order. So i'm using a signal as a setter.